### PR TITLE
Tidy up errors

### LINKED ISSUE #512: Tidy up Solidity custom errors, consider using libraries

### PR DESCRIPTION
**Pull request by ajsantander**
Fix #512

This PR reviews the overall usage of errors.

CommonErrors is removed and replaced with a bunch of libraries. Instead of inheriting from CommonErrors, you can just import the library and use it directly. I think it reads better.

Other errors are reviewed and semantically tuned.

Also found a logical error in `Ownable.renounceOwnership()` and fixed it.

### CODE REVIEW

File: packages/core-contracts/contracts/ownership/Ownable.sol [hunk 1]
```diff
@@ -24,22 +25,22 @@ contract Ownable is IOwnable, OwnableMixin, CommonErrors {
...
         if (newNominatedOwner == store.nominatedOwner) {
-            revert InvalidNomination(newNominatedOwner);
+            revert ChangeError.NoChange();
         }
 
         store.nominatedOwner = newNominatedOwner;
         emit OwnerNominated(newNominatedOwner);
     }
 
-    function renounceNomination() external override onlyOwner {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/ownership/Ownable.sol:38]: @i-stam I think I found an actual problem in the logic here.

The code that was in place semantically represents `removeNomination`, which the owner would be able to do.

`renounceNomination` should be done by whoever is nominated.

File: packages/core-contracts/contracts/proxy/Beacon.sol [hunk 2]
```diff
@@ -2,32 +2,33 @@
...
 
 import "./BeaconStorage.sol";
-import "../common/CommonErrors.sol";
 import "../ownership/OwnableMixin.sol";
 import "../utils/ContractUtil.sol";
+import "../interfaces/IBeacon.sol";
+import "../errors/AddressError.sol";
 
-contract Beacon is OwnableMixin, BeaconStorage, ContractUtil, CommonErrors {
+contract Beacon is IBeacon, OwnableMixin, BeaconStorage, ContractUtil {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/proxy/Beacon.sol:10]: Flyby fix, Beacon was not implementing IBeacon.

File: packages/core-contracts/contracts/token/ERC721.sol [hunk 3]
```diff
@@ -43,15 +40,17 @@ contract ERC721 is IERC721, IERC721Metadata, ERC721Storage, ContractUtil, Common
 
     function balanceOf(address holder) public view virtual override returns (uint) {
         if (holder == address(0)) {
-            revert InvalidAddress(holder);
+            revert AddressError.ZeroAddress();
         }
+
```
Comment (ajsantander) [inline packages/core-contracts/contracts/token/ERC721.sol:45]: Flyby line space fixes...

**Review by i-stam — APPROVED** [review]

File: packages/deployer/test/unit/internal/process-transactions.test.js [hunk 4]
```diff
@@ -69,9 +69,9 @@ describe('internal/process-transactions.js', function () {
       });
 
       it('registers a transaction in the deployment data', async function () {
-        const transaction = hre.deployer.deployment.general.transactions[transaction.hash];
-        equal(transaction.status, 'confirmed');
-        equal(transaction.description, 'test');
+        const tx = hre.deployer.deployment.general.transactions[transaction.hash];
```
Comment (ajsantander) [inline packages/deployer/test/unit/internal/process-transactions.test.js:72]: This was also fixed in another PR. Not sure how this made it to main.

Comment (mjlescano) [reply packages/deployer/test/unit/internal/process-transactions.test.js:72]: Probably a back merge of the `main` branch?

Comment (leomassazza) [reply packages/deployer/test/unit/internal/process-transactions.test.js:72]: The other fix is not merged yet (PR 513)

**Review by leomassazza — APPROVED** [review]

**Pull request merged**